### DISCUSSION TITLE
Question: What is the intended sharing process with upstream?

### DISCUSSION THREAD

**Discussion by GitMensch** [PowerShell/Win32-OpenSSH#2257] [General]
This is a follow-up to https://github.com/PowerShell/openssh-portable/issues/471 and moved here as requested by @bagajjal.

> Obviously many of the code changes would be relevant to be placed upstream - is there any plan to do send PRs?
> Even more obviously this repo tends to get outdated, what are the plans on pulling OpenSSH 9.6-9.8 here?
> 
> I thought to find the information in one of the repo's README or the Wiki - but had no luck with this.

https://github.com/openssh/openssh-portable/pulls?q=+is%3Apr+author%3Amanojampalam which was referenced in the issue above only includes old PRs,

Currently this repo lists "700 commits ahead, 231 commits behind" upstream.

https://github.com/PowerShell/Win32-OpenSSH/wiki/Project-Status references https://github.com/openssh/openssh-portable/pull/63 which was closed long time ago - this part should definitely be updated.

**Comment by maertendMSFT** [comment]
There are two questions here, lets address each individually.
1. When are we pulling in the latest updates from upstream OpenSSH?
- When OpenSSH puts out a release, we perform an analysis on whether or not Windows is impacted by the included changes.  If there have not been significant impacts to Windows, we do not pull in the update.  We know this is not a transparent process currently, and we are working on changing that within the internal requirements that we have.
        
        
2. Why don't we pull in our changes to upstream OpenSSH?
- This is our long-term goal, but there are still changes that need to be made on the Windows side before we are comfortable making the large changes to OpenSSH upstream. We are making progress on these items, inclusion in Windows Server 2025 by default is a big step in the right direction.  But, this is a multi-year initiative.
